js/map.js

```diff
@@ -73,49 +98,76 @@ function initialize() {
-          strokeWeight: 2,
-          fillColor: '#0000cc',
-          fillOpacity: 0.10,
-          map: map
-        });
-        if (piePoly) { /*Make code climate happy*/ }
-      }
-
-      //Draw links
-      if (results['router'] != undefined) {
-        var routerNode = nodeData[results['router']];
-        if (routerNode != undefined) {
-          var routerLink = [
-            { lat: results['latitude'], lng: results['longitude'] },
-            { lat: routerNode['latitude'], lng: routerNode['longitude'] },
-          ];
-
-          links[results['router'] + '-' + results['name']] = new google.maps.Polyline({
-            path: routerLink,
-            geodesic: true,
-            strokeColor: '#FF0000',
-            strokeOpacity: 1.0,
+        markers[results['name']]=newNode
+
+        //Draw cone
+        if (results['antennaCone'] != undefined) {
+          var antennaCone = results['antennaCone'];
+          var antennaDirection = results['antennaDirection'];
+          var antennaDistance = results['antennaDistance'];
+
+          var startArc = antennaDirection - (antennaCone / 2);
+          if (startArc < 0) startArc = startArc + 365;
+          var arcPts = drawArc(nodeLatLng, startArc, startArc + antennaCone, antennaDistance);
+          var piePoly = new google.maps.Polygon({
+            paths: [arcPts],
+            strokeColor: '#0000FF',
+            strokeOpacity: 0.2,
             strokeWeight: 2,
+            fillColor: '#0000cc',
+            fillOpacity: 0.10,
             map: map
           });
+          if (piePoly) { /*Make code climate happy*/ }
+        }
+      }
+    }
+    for (var key in data.nodeList) {
+      var results = data.nodeList[key];
+      if (results['status']!='invisible') {
+        //Draw links Router-AntennaRouter
+        if (results['router'] != undefined) {
+          var routerNode = nodeData[results['router']];
+          if (routerNode != undefined) {
+            var routerLink = [
+              { lat: results['latitude'], lng: results['longitude'] },
+              { lat: routerNode['latitude'], lng: routerNode['longitude'] },
+            ];
+
+            links[results['router'] + '-' + results['name']] = new google.maps.Polyline({
+              path: routerLink,
+              geodesic: true,
+              strokeColor: '#FF0000',
+              strokeOpacity: 1.0,
+              strokeWeight: 2,
+              map: map
+            });
+          }
+        }
+        //Draw links ssid station to ssid ap
+        if (results['mode'] == 'station' && results['ssid'] != undefined) {
+          var parentNode = nodeDataBySSID[results['ssid']];
+          if (parentNode != undefined) {
+            var parentLink = [
+              { lat: results['latitude'], lng: results['longitude'] },
+              { lat: parentNode['latitude'], lng: parentNode['longitude'] },
+            ];
+
+            links[parentNode['name'] + '-' + results['name']] = new google.maps.Polyline({
```

Comment:
When constructing a lookup key from parameters like this I usually prefer to use a function. Inline constructions like this are prone to typos and I've gotten annoyed debugging my own code when an attribute was undefined for some reason (assigned or referenced using the wrong key).